controllers/dnspolicy_controller.go

```diff
@@ -17,184 +17,11 @@ limitations under the License.
-			return ctrl.Result{Requeue: true}, err
-		} else if apierrors.IsNotFound(err) {
-			return ctrl.Result{}, err
-		}
-	}
-
-	specErr := r.reconcileResources(ctx, dnsPolicy, targetNetworkObject)
-
-	statusResult, statusErr := r.reconcileStatus(ctx, dnsPolicy, specErr)
-
-	if specErr != nil {
-		return ctrl.Result{}, specErr
-	}
-
-	return statusResult, statusErr
-}
-
-func (r *DNSPolicyReconciler) reconcileResources(ctx context.Context, dnsPolicy *v1alpha1.DNSPolicy, targetNetworkObject client.Object) error {
-	// reconcile based on gateway diffs
-	gatewayDiffObj, err := reconcilers.ComputeGatewayDiffs(ctx, r.Client(), dnsPolicy, targetNetworkObject)
-	if err != nil {
-		return err
-	}
-
-	if err = r.reconcileDNSRecords(ctx, dnsPolicy, gatewayDiffObj); err != nil {
-		return fmt.Errorf("error reconciling DNSRecords %w", err)
-	}
-
-	// set direct back ref - i.e. claim the target network object as taken asap
-	if err = r.TargetRefReconciler.ReconcileTargetBackReference(ctx, dnsPolicy, targetNetworkObject, dnsPolicy.DirectReferenceAnnotationName()); err != nil {
-		return fmt.Errorf("reconcile TargetBackReference error %w", err)
-	}
-
-	// set annotation of policies affecting the gateway
-	if err := r.TargetRefReconciler.ReconcileGatewayPolicyReferences(ctx, dnsPolicy, gatewayDiffObj); err != nil {
-		return fmt.Errorf("ReconcileGatewayPolicyReferences error %w", err)
-	}
-
-	return nil
-}
-
-func (r *DNSPolicyReconciler) deleteResources(ctx context.Context, dnsPolicy *v1alpha1.DNSPolicy, targetNetworkObject client.Object) error {
-	// delete based on gateway diffs
-	if err := r.deleteDNSRecords(ctx, dnsPolicy); err != nil {
-		return err
-	}
-
-	// remove direct back ref
-	if targetNetworkObject != nil {
-		if err := r.TargetRefReconciler.DeleteTargetBackReference(ctx, targetNetworkObject, dnsPolicy.DirectReferenceAnnotationName()); err != nil {
-			return err
-		}
-	}
-
-	gatewayDiffObj, err := reconcilers.ComputeGatewayDiffs(ctx, r.Client(), dnsPolicy, targetNetworkObject)
-	if err != nil {
-		return err
-	}
-
-	// update annotation of policies affecting the gateway
-	return r.TargetRefReconciler.ReconcileGatewayPolicyReferences(ctx, dnsPolicy, gatewayDiffObj)
-}
-
-func (r *DNSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	ok, err := kuadrantgatewayapi.IsGatewayAPIInstalled(mgr.GetRESTMapper())
-	if err != nil {
-		return err
-	}
-	if !ok {
-		r.Logger().Info("DNSPolicy controller disabled. GatewayAPI was not found")
-		return nil
-	}
-
-	gatewayEventMapper := mappers.NewGatewayEventMapper(
-		v1alpha1.NewDNSPolicyType(),
-		mappers.WithLogger(r.Logger().WithName("gateway.mapper")),
-		mappers.WithClient(mgr.GetClient()),
-	)
-
-	r.dnsHelper = dnsHelper{Client: r.Client()}
-	ctrlr := ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha1.DNSPolicy{}).
-		Owns(&kuadrantdnsv1alpha1.DNSRecord{}).
-		Watches(&gatewayapiv1.Gateway{}, handler.EnqueueRequestsFromMapFunc(gatewayEventMapper.Map))
-	return ctrlr.Complete(r)
-}
-
```

Comment:
doesn't seem to be much left in here, do we want to get rid entirely?

Reply:
In this PR i am deliberately not moving code that i don't need to so i don't have to deal with a review of existing code. I will be following this PR up with another to remove some of the existing controller files, this will have no functional changes, only moving stuff about to fit better with the new way of working.